lib/ansible/playbook/role_include.py

```diff
@@ -43,6 +43,11 @@ class IncludeRole(TaskInclude):
     circumstances related to the `- include_role: ...`
     """
 
+    BASE = set(['name', 'role'])
+    FROM_ARGS = set(['tasks_from', 'vars_from', 'defaults_from'])
+    OTHER_ARGS = set(['private', 'allow_duplicates'])
+    VALID_ARGS = frozenset(BASE.union(FROM_ARGS.union(OTHER_ARGS)))
```

Comment:
Strike ```coming from```, I see that there's a different reason for those in the code below. Still good to comment what the purpose of each is here.